src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -16,26 +16,42 @@ namespace Microsoft.Extensions.Caching.StackExchangeRedis;
 /// </summary>
 public class RedisCache : IDistributedCache, IDisposable
 {
+    // -- Explanation of why two kinds of SetScript are used --
+    // * Redis 2.0 had HSET key field value for setting individual hash fields,
+    // and HMSET key field value [field value ...] for setting multiple hash fields (against the same key)
+    // * Redis 4.0 observes this redundancy, and adds HSET key field value [field value ...],
+    // and also marks HMSET as deprecated, although it still works fine.
+    // But HSET doesn't allows multiple field/value pairs for Redis prior 4.0.
+
     // KEYS[1] = = key
     // ARGV[1] = absolute-expiration - ticks as long (-1 for none)
     // ARGV[2] = sliding-expiration - ticks as long (-1 for none)
     // ARGV[3] = relative-expiration (long, in seconds, -1 for none) - Min(absolute-expiration - Now, sliding-expiration)
     // ARGV[4] = data - byte[]
     // this order should not change LUA script depends on it
-    private const string SetScript = (@"
+    private const string SetScriptPerExtendedSetCommand = (@"
```

Comment:
@NickCraver @adityamandaleeka Was this supposed to be SetScript*Pre*ExtendedSetCommand? And was this meant to apply to what's currently called DeprecatedSetScript?

Reply:
@halter73 You're right, that's how I interpreted @NickCraver's comment too. I saw that it was renamed but didn't catch that it was applied to the newer one (in which case I guess "per" kinda makes sense :smile:).

Reply:
@halter73 yep, that's what I was pitching there :)

Reply:
I'll send a quick PR to rename them to SetScript and SetScriptPreExtendedSetCommand.

Reply:
https://github.com/dotnet/aspnetcore/pull/38928

Reply:
Thanks again for noticing that @halter73

Reply:
@adityamandaleeka, @halter73, thanks!